w10h02/test/pgdp/teams/UnitTest.java

```diff
@@ -47,4 +47,104 @@ public void optimalLineupAllAttackers() throws NoSuchFieldException, IllegalAcce
+                2, 2, 3);
+
+        var attackers = getField(lineup, "attackers");
+        var defenders = getField(lineup, "defenders");
+        var supporters = getField(lineup, "supporters");
+
+        assertEquals(2, attackers.size());
+        assertEquals(2, defenders.size());
+        assertEquals(3, supporters.size());
+
+        assertEquals(Set.of(eve, georg), attackers);
+        assertEquals(Set.of(jasmine, anton), defenders);
+        assertEquals(Set.of(jakob, yassine, faid), supporters);
+    }
+
+    @Test
+    @DisplayName("Soll funktionieren wenn defenders \"groß\" ist")
+    public void optimalLineUpLargeSubsetSize() throws NoSuchFieldException, IllegalAccessException {
+        var lineup = Lineup.computeOptimalLineup(Set.of(
+                jasmine, yassine, eve, anatoly, anton, cedric, faid, jakob, georg, hanna, jakov,
+                        felix, jan, shrek), 0, 10, 0);
+
+        var attackers = getField(lineup, "attackers");
+        var defenders = getField(lineup, "defenders");
+        var supporters = getField(lineup, "supporters");
+
+        assertEquals(0, attackers.size());
+        assertEquals(10, defenders.size());
+        assertEquals(0, supporters.size());
+
+        assertEquals(Set.of(eve), attackers);
+        assertEquals(Set.of(jan), defenders);
+        assertEquals(Set.of(jakob), supporters);
+    }
+
+    @Test
+    @DisplayName("OPTIONAL - berechnet großes Lineup und printed die Laufzeit. Timeout nach 30 sekunden, " +
+            "allerdings wird der test nur als failed gewertet wenn er nicht timeouted aber das ergebnis falsch ist.")
+    public void laufzeitFunTest() throws InterruptedException {
+        long t1 = System.currentTimeMillis();
+        ExecutorService es = Executors.newSingleThreadExecutor();
+        es.submit(() -> {
+            var lineup = Lineup.computeOptimalLineup(Set.of(
+                    jasmine, yassine, eve, anatoly, anton, cedric, faid, jakob, georg, hanna,
+                    felix, jan, zeynep, shrek), 4, 4, 4);
+
+
+            try {
+                var attackers = getField(lineup, "attackers");
+                var defenders = getField(lineup, "defenders");
+                var supporters = getField(lineup, "supporters");
+
+                assertEquals(4, attackers.size());
+                assertEquals(4, defenders.size());
+                assertEquals(4, supporters.size());
+
+                assertEquals(Set.of(hanna, eve, georg, shrek), attackers);
+                assertEquals(Set.of(anatoly, anton, jan, jasmine), defenders);
+                assertEquals(Set.of(yassine, faid, felix, jakob), supporters);
+
+            } catch (NoSuchFieldException | IllegalAccessException e) {
+                e.printStackTrace();
+            }
+        });
+
+        es.shutdown();
+        if(!es.awaitTermination(2, TimeUnit.SECONDS)) {
```

Comment:
Müsste hier nicht 
```java
es.awaitTermination(30, TimeUnit.SECONDS)
```
stehen?

In deiner Beschreibung steht nämlich `30` Sekunden.

Reply:
Oh ja, klar. Das hatte ich zum testen auf 2 sekunden runtergeschraubt um zu checken ob es auch wirklich timeouted...